### DISCUSSION TITLE
$dispatch('input', ...) does nothing

### DISCUSSION THREAD

**Discussion by Sch-Tim** [livewire/livewire#5051] [1. Help] [answered]
Since I don't think, that this is an error but I'm just doing something wrong, I used the "help" category. 

I'm trying to implement the ToastUI editor in my Laravel-Livewire application with the following code:  

**editor.blade.php (a Laravel component)**
```
<div>
    <div x-data="{
    init() {
      const editor = new toastui.Editor({
        el: document.querySelector('#editor_{{ $divIdSuffix }}'),
        events: {
            change: onChange
        },
        previewStyle: 'vertical',
        height: '500px',
        initialValue: 'TEST'
      });

      const viewer = toastui.Editor.factory({
        el: document.querySelector('#viewer_{{ $divIdSuffix }}'),
        viewer: true,
        height: '500px',
      });

      function onChange(){
        $dispatch('input', editor.getHTML()); **// does nothing**
        console.log(editor.getHTML()); **// works**
        }
      }
    }">
        <!-- Editor -->
        <h2>Editor</h2>
        <div id="editor_{{$divIdSuffix}}"></div>
        <!-- Viewer Using Editor -->
        <h2>Viewer</h2>
        <div id="viewer_{{$divIdSuffix}}"></div>
    </div>
</div>
```

**edit-text-block-component.blade.php (a Livewire component)**
```
   <x-inputs.editor wire:model="test"/>
   <input wire:model="test" />
```

**EditTextBlockComponent.php**
    public function updatedTest($value)
    {
        echo("");
    }

To test my implementation, I just added an echo() to put a breakpoint there. The problem is, that although the event fires (the console.log line shows the updated input), the updatedTest() function is not called and the `<input>` is not updated. 

I also tried to call this.$dispatch which gives me a `this.$dispatch is not a function` error.

**Comment by Martleby** [comment, marked as answer]
If I'm understanding this correctly, when the editor text changes you want to set a property (input?) on your Livewire component and have it update?

If so, you need to change this line:
```
$dispatch('input', editor.getHTML()); **// does nothing**
```

to

```
$wire.set('input', editor.getHTML()) //Assuming that there is a property on your model called input
```

or, if you actually want to call the method updatedTest on you component

```
$wire.updatedTest(editor.getHTML())
```

$wire is some magic that allows you easily interact with Livewire from Alpine, some useful info here: https://laravel-livewire.com/docs/2.x/alpine-js

If I've completely misunderstood what you're trying to do, let me know and I'll see if I can help further :)

Cheers,
Martin.

**Reply by Sch-Tim** [reply]
Hello Martin,
thanks for your answer. I think you didn't understand me correctly.
I try to realize the example explained here: https://laravel-livewire.com/docs/2.x/alpine-js#forwarding-wire-model-input-events

If I understood this correctly it's possible to use the Wire:model functionality for custom Inputs by dispatching an Input Event from the child component.

Of course I could just use custom events to update some variable but I want to debounce my input and make use of wire:model.

Greets
Tim

**Reply by Martleby** [reply]
Hey Tim,

Ah, I see! 

I notice that you specify:

```
<x-inputs.editor wire:model="test"/>
```
in edit-text-block-component.blade.php, but in editor.blade.php you don't appear to echo out the $attributes anywhere. You might want to try adding `{{$attributes}}` somewhere on the div with your x-data, like:

```
 <div 
  {{$attributes}} //Add this
   x-data="{
    init() {
      const editor = new toastui.Editor({
        el: document.querySelector('#editor_{{ $divIdSuffix }}'),
        events: {
            change: onChange
        },
      ......
```

Cheers,
Martin.

**Reply by Sch-Tim** [reply]
Thank you so much. 
I knew it had to be something simple. 

I totally overlooked that line in the example. 

Have a nice week!
